cmd/controller/main.go

```diff
@@ -102,7 +102,7 @@ func main() {
 
 	// Build all of our controllers, with the clients constructed above.
 	// Add new controllers to this array.
-	controllers := []*controller.Impl{
+	controllers := [...]*controller.Impl{
```

Comment:
If we're only concerned about the upper bound then you could just have this line be`[numControllers]*controller.Impl` 

Then you can drop the compile time assertion below - since the compiler will complain if this array has more that `numController` elements

ie. https://play.golang.org/p/Q3Coj5TecnI

Reply:
But it won't if there're less: https://github.com/knative/eventing/pull/1112#discussion_r279440195